src/inc/shash.h

```diff
@@ -479,6 +498,10 @@ class SHash : public TRAITS
             return *(const Iterator*)this;
         }
 
+        KeyIterator()
```

Comment:
Can you comment on these changes?  I'm a little suspicious of them because it looks like they allow these structures to not care about the SHash that they're operating on.